### PR TITLE
Gb3 profile overview

### PR DESCRIPTION
**Pull request by kristina-kolosek**
Please take a look @salac-kolosek @kris-kolosek

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  private
+
+  def profile_params
+    params.require(:profile).permit(:full_name, :slug, :biography)
+  end
+
+  def user_params
+    params.require(:profile).permit(user: [:email])
+  end
+
```
Comment (sniffler) [inline app/controllers/profiles_controller.rb]: Extra empty line detected at class body end.

File: app/assets/stylesheets/profiles.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+}
+
+.cover-placeholder {
+  background: 50% no-repeat #fafafa;
+  background-size: cover;
+  background: image-url("user-cover.png");
+  background-color: lightseagreen;
```
Comment (sniffler) [inline app/assets/stylesheets/profiles.scss]: Color `lightseagreen` should be written in hexadecimal form as `#20b2aa`<br>Color literals like `lightseagreen` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/profiles.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  width: 100%;
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+}
+
+.cover-placeholder {
+  background: 50% no-repeat #fafafa;
+  background-size: cover;
+  background: image-url("user-cover.png");
```
Comment (sniffler) [inline app/assets/stylesheets/profiles.scss]: Property `background` already defined on line 13

File: app/assets/stylesheets/profiles.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+.profile-page {
+  width: 100%;
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+}
+
+.cover-placeholder {
+  background: 50% no-repeat #fafafa;
+  background-size: cover;
```
Comment (sniffler) [inline app/assets/stylesheets/profiles.scss]: Properties should be ordered background, background-color, background-size, border-radius, height, margin-bottom, width

File: app/assets/stylesheets/profiles.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+.profile-page {
+  width: 100%;
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+}
+
+.cover-placeholder {
+  background: 50% no-repeat #fafafa;
```
Comment (sniffler) [inline app/assets/stylesheets/profiles.scss]: Color literals like `#fafafa` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/profiles.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// Place all the styles related to the profiles controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+
+.profile-page {
+  width: 100%;
```
Comment (sniffler) [inline app/assets/stylesheets/profiles.scss]: Properties should be ordered display, flex-direction, padding, width

File: app/assets/stylesheets/partials/_index.scss [hunk 7]
```diff
@@ -80,6 +93,10 @@
     color: #fff;
     text-shadow: 0 -1px 0 rgba(0,0,0,.1);
     font-size: 13px;
+
+    &.--blue {
+      background: linear-gradient(#4ab6f0,#2fa5e4 60%,#2fa5e4 90%,#38a9e5);
```
Comment (sniffler) [inline app/assets/stylesheets/partials/_index.scss]: Color literals like `#4ab6f0` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>Commas in function arguments should be followed by one space<br>Color literals like `#2fa5e4` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>Color literals like `#38a9e5` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/partials/_index.scss [hunk 8]
```diff
@@ -67,6 +69,17 @@
     white-space: nowrap;
   }
 
+  .chevron-icon {
+    margin: 10px;
+  }
+
+  .header-breadcrumbs {
+    display: flex;
```
Comment (sniffler) [inline app/assets/stylesheets/partials/_index.scss]: Properties should be ordered align-items, display, flex-direction, justify-content

File: app/assets/stylesheets/components/profile-form.scss [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    &.--area {
+      min-height: 120px;
+    }
+  }
+
+  .helper-text {
+    font-size: 13px;
+    margin-top: 5px;
+    color: #888;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Color literals like `#888` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/profile-form.scss [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    width: 100%;
+    font-family: inherit;
+
+    &.--area {
+      min-height: 120px;
+    }
+  }
+
+  .helper-text {
+    font-size: 13px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Properties should be ordered color, font-size, margin-top

File: app/assets/stylesheets/partials/_index.scss [hunk 11]
```diff
@@ -67,6 +69,17 @@
...
   }
 
+  .chevron-icon {
+    margin: 10px;
+  }
+
+  .header-breadcrumbs {
+    align-items: center;
+    display: flex;
+    justify-content: flex-start;
```
Comment (sniffler) [inline app/assets/stylesheets/partials/_index.scss]: Properties should be ordered align-items, display, flex-direction, justify-content

File: app/assets/stylesheets/components/profile-form.scss [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    padding: 10px 10px 10px 35px;
+    width: 100%;
+
+    &.--area {
+      min-height: 120px;
+    }
+  }
+
+  .helper-text {
+    color: #888;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Color literals like `#888` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/profile-form.scss [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  .field {
+    border: 1px solid #dae1e3;
+    border-radius: 4px;
+    color: #4b5b62;
+    display: block;
+    font-family: inherit;
+    font-size: 18px;
+    font-weight: 100;
+    height: 44px;
+    padding: 10px 10px 10px 35px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Shorthands of length `4` are not allowed. Value was `10px 10px 10px 35px`

File: app/assets/stylesheets/components/profile-form.scss [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    font-size: 14px;
+    font-weight: bold;
+    line-height: 18px;
+    margin: 0px;
+  }
+
+  .field {
+    border: 1px solid #dae1e3;
+    border-radius: 4px;
+    color: #4b5b62;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Color literals like `#4b5b62` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/profile-form.scss [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+  .lbl {
+    font-size: 14px;
+    font-weight: bold;
+    line-height: 18px;
+    margin: 0px;
+  }
+
+  .field {
+    border: 1px solid #dae1e3;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Color literals like `#dae1e3` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/profile-form.scss [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+  text-align: left;
+  max-width: 700px;
+  position: relative;
+  width: 80%;
+
+  .lbl {
+    font-size: 14px;
+    font-weight: bold;
+    line-height: 18px;
+    margin: 0px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: `0px` should be written without units as `0`

File: app/assets/stylesheets/components/profile-form.scss [hunk 17]
```diff
@@ -0,0 +1,38 @@
+.profile-form {
+  display: flex;
+  flex-direction: column;
+  text-align: left;
```
Comment (sniffler) [inline app/assets/stylesheets/components/profile-form.scss]: Properties should be ordered display, flex-direction, max-width, position, text-align, width

File: app/assets/stylesheets/components/auth-form.scss [hunk 18]
```diff
@@ -32,40 +32,40 @@
...
   overflow: hidden;
   padding: 0 12px;
   text-align: center;
-  -webkit-box-shadow: 0 1px 0 rgba(0,0,0,.12);
-  box-shadow: 0 1px 0 rgba(0,0,0,.12);
-  color: #fff;
-  fill: #fff;
   text-shadow: 0 -1px 0 rgba(0,0,0,.1);
   transition: none !important;
+  -webkit-box-shadow: 0 1px 0 rgba(0,0,0,.12);
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Avoid vendor prefixes.<br>Color literals like `rgba(0, 0, 0, 0.12)` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>Commas in function arguments should be followed by one space<br>`.12` should be written with a leading zero as `0.12`

File: app/assets/stylesheets/components/auth-form.scss [hunk 19]
```diff
@@ -32,40 +32,40 @@
...
     width: 100%;
-    font-family: inherit;
   }
 }
 
 .button {
   border-radius: 5px;
+  box-shadow: 0 1px 0 rgba(0,0,0,.12);
+  color: #fff;
+  fill: #fff;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Color literals like `#fff` should only be used in variable declarations; they should be referred to via variable everywhere else.

**Comment by kolosek** [comment]
@kristina-kolosek Please connect codeclimate also, and read sniffler comments first, before asking others to make comments, and do corrections for those.

File: app/assets/stylesheets/partials/_index.scss [hunk 20]
```diff
@@ -76,10 +90,14 @@
...
     border: 0;
-    fill: #fff;
-    color: #fff;
+    fill: $white;
+    color: $white;
     text-shadow: 0 -1px 0 rgba(0,0,0,.1);
     font-size: 13px;
+
+    &.--blue {
+      background: linear-gradient($light-blue,$dark-blue 60%,$dark-blue 90%,$medium-blue);
```
Comment (sniffler) [inline app/assets/stylesheets/partials/_index.scss]: Commas in function arguments should be followed by one space

File: app/assets/stylesheets/components/auth-form.scss [hunk 21]
```diff
@@ -32,43 +33,43 @@
...
-  box-shadow: 0 1px 0 rgba(0,0,0,.12);
-  color: #fff;
-  fill: #fff;
   text-shadow: 0 -1px 0 rgba(0,0,0,.1);
   transition: none !important;
+  -webkit-box-shadow: 0 1px 0 $box-shadow;
 
    &.--blue {
-     background: linear-gradient(#4ab6f0,#2fa5e4 60%,#2fa5e4 90%,#38a9e5);
+     background: linear-gradient($light-blue,$dark-blue 60%,$dark-blue 90%,$medium-blue);
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Commas in function arguments should be followed by one space

File: app/assets/stylesheets/components/auth-form.scss [hunk 22]
```diff
@@ -32,43 +33,43 @@
...
   overflow: hidden;
   padding: 0 12px;
   text-align: center;
-  -webkit-box-shadow: 0 1px 0 rgba(0,0,0,.12);
-  box-shadow: 0 1px 0 rgba(0,0,0,.12);
-  color: #fff;
-  fill: #fff;
   text-shadow: 0 -1px 0 rgba(0,0,0,.1);
   transition: none !important;
+  -webkit-box-shadow: 0 1px 0 $box-shadow;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Avoid vendor prefixes.

File: app/assets/stylesheets/components/auth-form.scss [hunk 23]
```diff
@@ -32,43 +33,43 @@
...
     height: 44px;
+    padding: 10px 10px 10px 35px;
     width: 100%;
-    font-family: inherit;
   }
 }
 
 .button {
   border-radius: 5px;
+  box-shadow: 0 1px 0 $box-shadow;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss:57]: Properties should be ordered border-radius, -webkit-box-shadow, box-shadow, color, fill, font-size, font-weight, height, line-height, overflow, padding, text-align, text-shadow, transition

File: app/assets/stylesheets/components/auth-form.scss [hunk 24]
```diff
@@ -32,43 +33,43 @@
...
+    border: 1px solid $field-border;
     border-radius: 4px;
-    color: #4b5b62;
+    color: $input-color;
     display: block;
+    font-family: inherit;
+    font-size: 18px;
+    font-weight: 100;
     height: 44px;
+    padding: 10px 10px 10px 35px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Shorthands of length `4` are not allowed. Value was `10px 10px 10px 35px`

File: app/assets/stylesheets/components/auth-form.scss [hunk 25]
```diff
@@ -32,43 +33,43 @@
 
   .lbl {
     font-size: 14px;
-    margin: 0px;
-    line-height: 18px;
     font-weight: bold;
+    line-height: 18px;
+    margin: 0px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: `0px` should be written without units as `0`

File: app/assets/stylesheets/components/auth-form.scss [hunk 26]
```diff
@@ -1,26 +1,27 @@
...
   width: 100%;
-  position: relative;
 
   .avatar-placeholder {
-    height: 80px;
-    width: 80px;
+    background-color: #C6DAE3;
+    border: 3px solid white;
     border-radius: 50%;
+    height: 80px;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Properties should be ordered background-color, border, border-radius, -webkit-box-shadow, height, left, position, top, width

File: app/assets/stylesheets/components/auth-form.scss [hunk 27]
```diff
@@ -1,26 +1,27 @@
...
+  position: relative;
   text-align: left;
   width: 100%;
-  position: relative;
 
   .avatar-placeholder {
-    height: 80px;
-    width: 80px;
+    background-color: #C6DAE3;
+    border: 3px solid white;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Color `white` should be written in hexadecimal form as `#ffffff`<br>Color literals like `white` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/auth-form.scss [hunk 28]
```diff
@@ -1,26 +1,27 @@
...
   padding: 45px;
+  position: relative;
   text-align: left;
   width: 100%;
-  position: relative;
 
   .avatar-placeholder {
-    height: 80px;
-    width: 80px;
+    background-color: #C6DAE3;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Color literals like `#C6DAE3` should only be used in variable declarations; they should be referred to via variable everywhere else.<br>Color `#C6DAE3` should be written as `#c6dae3`

File: app/assets/stylesheets/base/colors.scss [hunk 29]
```diff
@@ -0,0 +1,9 @@
+$white: #fff;
+$light-blue: #4ab6f0;
+$medium-blue: #38a9e5;
+$dark-blue: #2fa5e4;
+$box-shadow: rgba(0, 0, 0, 0.12);
+$field-border: #dae1e3;
+$input-color: #4b5b62;
+$helper-text: #888;
+$avatar-background: #C6DAE3;
```
Comment (sniffler) [inline app/assets/stylesheets/base/colors.scss]: Color `#C6DAE3` should be written as `#c6dae3`

File: app/assets/stylesheets/components/auth-form.scss [hunk 30]
```diff
@@ -1,26 +1,27 @@
+@import "base/colors";
 
 .auth-form {
+  background: #f8fbfd;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Color literals like `#f8fbfd` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/components/auth-form.scss [hunk 31]
```diff
@@ -32,43 +33,43 @@
...
     height: 44px;
+    padding: 10px 10px 35px;
     width: 100%;
-    font-family: inherit;
   }
 }
 
 .button {
   border-radius: 5px;
+  -webkit-box-shadow: 0 1px 0 $box-shadow;
```
Comment (sniffler) [inline app/assets/stylesheets/components/auth-form.scss]: Avoid vendor prefixes.

**Comment by kristina-kolosek** [comment]
@kolosek I checked and code climate webhook is setup (same config like for sniffler), but for some reason it's not functioning well. I'm searching for a solution.

I fixed sniffler comments except for this one that keeps appearing even if all the CSS attributes are sorted properly.


Meanwhile, @salac-kolosek and @kris-kolosek  can now check the PR

File: app/controllers/application_controller.rb [hunk 32]
```diff
@@ -5,4 +5,10 @@ class ApplicationController < ActionController::Base
 
   def index
   end
+
+  def require_admin!
```
Comment (salac-kolosek) [inline app/controllers/application_controller.rb]: before_action :authenticate_user! :)

**Review by salac-kolosek — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 32 above)
Comment (kristina-kolosek) [reply app/controllers/application_controller.rb]: require_admin is for authorization check, used for checking whether user is admin, for pages like listing team members

File: app/controllers/application_controller.rb [hunk 33]
```diff
@@ -5,4 +5,5 @@ class ApplicationController < ActionController::Base
 
   def index
   end
+
```
Comment (sniffler) [inline app/controllers/application_controller.rb]: Extra empty line detected at class body end.

**Comment by kolosek** [comment]
@kristina-kolosek Sale connected codeclimate before, he could check it out as well.

**Pull request merged**